app/javascript/packs/CommentTextArea/CommentTextArea.jsx

```diff
@@ -35,9 +58,24 @@ export const CommentTextArea = ({ vanillaTextArea }) => {
     }
   };
 
+  const setPasteElement = usePasteImage({
+    onPaste: handleImagePasted(
+      handleImageSuccess(textAreaRef),
+      handleImageFailure,
+    ),
+  });
+
+  const onchangeEvent = () => {
+    if (textAreaRef.current) {
+      setPasteElement(textAreaRef.current);
+    }
+  };
+
   return (
     <div className="w-100 relative">
       <AutocompleteTriggerTextArea
+        ref={textAreaRef}
+        onChange={onchangeEvent}
```

Comment:
The `onChange` callback will be invoked every character that the user types, but we only need to `setPasteElement` once. `onChange` is also only called once the user starts typing, so if you click into a text area and immediately paste an image it currently isn't working.

Instead of using the `onChange` callback, how about setting the paste element in a `useLayoutEffect`? e.g.

```js
  useLayoutEffect(() => {
    if (textAreaRef.current) {
      setPasteElement(textAreaRef.current);
    }
  }, [setPasteElement]);
```

(`useLayoutEffect` as opposed to `useEffect` so that the inner text area has fully rendered by the time it runs)

Reply:
Of course, I will make the respective changes.
A new thing I learn :smile: